scripts/egp39/previewProposal.ts

```diff
@@ -0,0 +1,306 @@
+  const callDataChangeVaultStatusHdMigrationRewardsVault =
+    coreVotingInterface.encodeFunctionData("changeVaultStatus", [
+      hdMigrationRewardsVaultAddress,
+      true,
+    ]);
+
+  // Add the hdMigrationLinearVestingVault to the approved vault list in the CoreVoting contract.
+  const callDataChangeVaultStatusHdMigrationLinearVestingVault =
+    coreVotingInterface.encodeFunctionData("changeVaultStatus", [
+      hdMigrationLinearVestingVaultAddress,
+      true,
+    ]);
+
+  // Add the hdLockingVault to the approved vault list in the gscCoreVoting contract.
+  const callDataChangeVaultStatusHdGscVault =
+    coreVotingInterface.encodeFunctionData("changeVaultStatus", [
+      hdGscVaultAddress,
+      true,
+    ]);
+
+  // Set approvals for the hdLockingVault from the treasury for HD tokens.
+  const treasuryInterface = new ethers.utils.Interface(Treasury__factory.abi);
+  console.log("hdTokenAddress", hdTokenAddress);
+  console.log("hdLockingVaultAddress", hdLockingVaultAddress);
+  const callDataTreasuryApproveHdLockingVault =
+    treasuryInterface.encodeFunctionData("approve", [
+      hdTokenAddress,
+      hdLockingVaultAddress,
+      MAX_UINT_256,
+    ]);
+  const callDataTreasuryApproveHdMigrationRewardsVault =
+    treasuryInterface.encodeFunctionData("approve", [
+      hdTokenAddress,
+      hdMigrationRewardsVaultAddress,
+      MAX_UINT_256,
+    ]);
+
+  const callDataTreasuryApproveHdMigrationLinearVestingVault =
+    treasuryInterface.encodeFunctionData("approve", [
+      hdTokenAddress,
+      hdMigrationLinearVestingVaultAddress,
+      MAX_UINT_256,
+    ]);
+  const callDataTreasuryApproveHdGscVault =
+    treasuryInterface.encodeFunctionData("approve", [
+      hdTokenAddress,
+      hdGscVaultAddress,
+      MAX_UINT_256,
+    ]);
+
+  // Convert ELFI -> HD tokens for Delv, Foundation, Treasury
+  const signer = await getSigner();
+  if (!signer) {
+    throw new Error("Signer not found");
+  }
+  const tokenContract = ERC20PermitWithMint__factory.connect(
+    tokenAddress,
+    signer
+  );
+
+  const delvBalance = await tokenContract.balanceOf(delvWalletAddress);
+  const foundationBalance = await tokenContract.balanceOf(
+    foundationWalletAddress
+  );
+  const treasuryBalance = await tokenContract.balanceOf(treasuryAddress);
+
+  const callDataBurnDelvBalance = elfiTokenInterface.encodeFunctionData(
+    "burn",
+    [delvWalletAddress, delvBalance]
+  );
+  const callDataBurnFoundationBalance = elfiTokenInterface.encodeFunctionData(
+    "burn",
+    [foundationWalletAddress, foundationBalance]
+  );
+  const callDataBurnTreasuryBalance = elfiTokenInterface.encodeFunctionData(
+    "burn",
+    [treasuryAddress, treasuryBalance]
+  );
+
+  const hdTokenInterface = new ethers.utils.Interface(
+    ERC20PermitWithMint__factory.abi
+  );
+
+  const callDataMintDelvBalance = hdTokenInterface.encodeFunctionData("mint", [
```

Comment:
Rather than minting, we should just use the initial allocation of tokens. 

The flow that I'm envisioning is:

1. Deploy the HD token.
2. Send HD tokens to the treasury.
3. The treasury approves the migration vault.
4. The treasury burns the ELFI tokens from DELV, the foundation, and the treasury.
5. The treasury transfers HD to DELV and the foundation (note, not to itself because it already has it) to compensate them for the burned ELFI.